src/transformers/typescript.ts

```diff
@@ -175,11 +312,192 @@ const transformer: Transformer<Options.Typescript> = ({
+  }
+
+  tsconfigFile = isAbsolute(tsconfigFile)
+    ? tsconfigFile
+    : join(basePath, tsconfigFile);
+
+  basePath = dirname(tsconfigFile);
+
+  const { error, config } = ts.readConfigFile(tsconfigFile, ts.sys.readFile);
+
+  if (error) {
+    throw new Error(formatDiagnostics(error, basePath));
+  }
+
+  // Do this so TS will not search for initial files which might take a while
+  config.include = [];
+
+  let { errors, options } = ts.parseJsonConfigFileContent(
+    config,
+    ts.sys,
+    basePath,
+    compilerOptionsJSON,
+    tsconfigFile,
+  );
+
+  // Filter out "no files found error"
+  errors = errors.filter((d) => d.code !== 18003);
+
+  return { errors, options };
+}
+
+async function mixedImportsTranspiler({
+  content,
+  filename = 'source.svelte',
+  markup,
+  options = {},
+  attributes,
+  compilerOptions,
+  basePath,
+}: InternalTransformerOptions) {
+  const sourceMapChain = createSourceMapChain({
+    filename,
+    content,
+    compilerOptions,
+  });
+
+  const injectedCode = injectVarsToCode({
+    content,
+    markup,
+    filename,
+    attributes,
+    sourceMapChain,
+  });
+
+  const { transpiledCode, sourceMapText, diagnostics } = transpileTs({
+    code: injectedCode,
+    fileName: `${filename}.injected.ts`,
+    basePath,
+    options,
+    compilerOptions,
+  });
+
+  if (sourceMapChain) {
+    const fname = `${filename}.transpiled.js`;
+
+    sourceMapChain.content[fname] = transpiledCode;
+    sourceMapChain.sourcemaps[fname] = JSON.parse(sourceMapText);
+  }
+
+  const code = stripInjectedCode({
+    transpiledCode,
+    markup,
+    filename,
+    sourceMapChain,
+  });
+
+  // Sorcery tries to load the code/map from disk if it's empty,
+  // prevent that because it would try to load inexistent files
+  // https://github.com/Rich-Harris/sorcery/issues/167
+  if (!code) {
+    return { code, diagnostics };
+  }
+
+  const map = await concatSourceMaps({
+    filename,
+    markup,
+    sourceMapChain,
+  });
+
   return {
     code,
     map,
     diagnostics,
   };
+}
+
+async function simpleTranspiler({
+  content,
+  filename = 'source.svelte',
+  options = {},
+  compilerOptions,
+  basePath,
+}: InternalTransformerOptions) {
+  const { transpiledCode, sourceMapText, diagnostics } = transpileTs({
+    code: content,
+    transformers: { before: [importTransformer] },
+    fileName: filename,
+    basePath,
+    options,
+    compilerOptions,
+  });
+
+  return {
+    code: transpiledCode,
+    map: sourceMapText,
+    diagnostics,
+  };
+}
+
+const transformer: Transformer<Options.Typescript> = async ({
+  content,
+  filename = 'source.svelte',
+  markup,
+  options = {},
+  attributes,
+}) => {
+  const basePath = process.cwd();
+
+  filename = isAbsolute(filename) ? filename : resolve(basePath, filename);
+
+  const compilerOptions = getCompilerOptions({ filename, options, basePath });
+  const canUseMixedImportsTranspiler = +pkg.version.split('.')[1] >= 39;
```

Comment:
Would it make sense to check for the major segment as well? Conceptually, just checking for the minor segment seems weird, albeit it's probably good enough 🤔